packages/mobile/src/transactions/saga.ts

```diff
@@ -149,8 +152,37 @@ function* refreshRecentTxRecipients() {
   yield put(updateRecentTxRecipientsCache(recentTxRecipientsCache))
 }
 
+function* addProfile(transaction: TransactionFeedFragment) {
+  const profiles = yield select(valoraRecipientCacheSelector)
+  // @ts-ignore transaction must have address because it is a TokenTransfer
+  const address = transaction.address
+  if (!profiles[address]) {
+    const newProfile: AddressToRecipient = {}
+    if (transaction.type === TokenTransactionType.Received) {
+      const info = yield call(getProfileInfo, address)
+      newProfile[address] = {
+        address,
+        name: info?.name,
+        thumbnailPath: info?.thumbnailPath,
+      }
+
+      yield put(updateValoraRecipientCache(newProfile))
+      Logger.info(TAG, `added ${newProfile} to valoraRecipientCache`)
+    }
+  }
+}
+
+function* addRecipientProfiles({ transactions }: NewTransactionsInFeedAction) {
+  yield all(
+    transactions
+      .filter((trans) => trans.__typename === 'TokenTransfer')
+      .map((trans) => call(addProfile, trans))
```

Comment:
I get an error when I try to use `forEach` here. I'm guessing it's because `forEach` doesn't return an array while `map` does, and something needs to be passed to the `all()`?